dataclients/kubernetes/kube.go

```diff
@@ -692,6 +704,13 @@ func (c *Client) ingressToRoutes(items []*ingressItem) ([]*eskip.Route, error) {
 			// currently handled as mandatory
 			host := []string{"^" + strings.Replace(rule.Host, ".", "[.]", -1) + "$"}
 
+			// add extra routes from optional annotation
+			for _, route := range extraRoutes {
+				route.HostRegexps = host
+				route.Id = routeID("", route.Id, rule.Host, "", "")
```

Comment:
i would keep the namespace in the route id.

since a single route without an id is a valid eskip expression, i would try to be more specific when generating this id here. A single route expression looks like this (no id and no semicolon):

```
Method("OPTIONS") -> setResponseHeader("Foo", "bar") -> status(200) -> <shunt>
```

eskip.Parse will return a slice with one route that has an empty Id.

How about generating the route id here so:

```
routeID(namespace, name + "_" + route.Id + "_" + indexString, rule.Host, "", "")
```

where the indexString comes from the index in the enclosing loop. Would it work?

If yes, then of course it would be nice the create a new function, s.g. like routeIDForCustom() or a similar name where the args are more expressive regarding what the function does.

Reply:
Yeah I thought about this but thought having the host name there would be enough uniqueness. Still it makes sense to have more context in the id for debugging. I have done something like you suggested.